src/garage/_functions.py

```diff
@@ -159,3 +159,43 @@ def log_performance(itr, batch, discount, prefix='Evaluation'):
             tabular.record('SuccessRate', np.mean(success))
 
     return undiscounted_returns
+
+
+def flatten_if_unflattened(observation_space, observation):
+    """Flattens observation to a numpy.ndarray if it is not.
```

Comment:
This isn't very clear, at least to me. In numpy-speak, flattened would mean `ndarray.ndim == 1`, but it looks like all this function does is ensure `isinstance(observation, np.ndarray)`. Basically, the `ndim` of the returned array is unconstrained, so "flattened" doesn't really mean what it usually means.